### PR TITLE
Refactor Config API and Settings pages for Reuse and Consistency

### PR DESCRIPTION
**Pull request by debanjum**
### Major
- Reuse get config data logic across config pages on web client
- Make config api endpoint urls consistent
- Rename API path /api/config to /api/configure
- Move Web client settings page to be under `/settings` from the previous `/config` url path
- Move Desktop settings page to be under `/settings` from the previous `/config` url path

### Minor
- Pass isMobileWidth prop to SidePanel via chat share interface
- Turn prettier off instead of throwing error for now
- Do no explicitly add line-clamp plugin as it's in Tailwind by default

### CODE REVIEW

File: src/interface/web/app/components/navMenu/navMenu.tsx [hunk 1]
```diff
@@ -128,7 +128,7 @@ export default function NavMenu(props: NavMenuProps) {
                                 <DropdownMenuSeparator />
                                 <DropdownMenuLabel>Profile</DropdownMenuLabel>
                                 <DropdownMenuItem>
-                                    <Link href="/config">Settings</Link>
+                                    <Link href="/configure">Configure</Link>
```
Comment (sabaimran) [inline src/interface/web/app/components/navMenu/navMenu.tsx:131]: I prefer settings for the settings page. Follows more standard UX nomenclature.

File: src/interface/web/app/components/navMenu/navMenu.tsx [hunk 2]
```diff
@@ -128,7 +128,7 @@ export default function NavMenu(props: NavMenuProps) {
                                 <DropdownMenuSeparator />
                                 <DropdownMenuLabel>Profile</DropdownMenuLabel>
```
Comment (sabaimran) [inline src/interface/web/app/components/navMenu/navMenu.tsx:129]: (For later) We should replace this with the user's avatar.

File: src/interface/web/app/components/navMenu/navMenu.tsx [hunk 3]
```diff
@@ -172,8 +172,8 @@ export default function NavMenu(props: NavMenuProps) {
                                 {userData &&
                                     <>
                                         <MenubarItem>
-                                            <Link href="/config">
-                                                Settings
+                                            <Link href="/configure">
+                                                Configure
```
Comment (sabaimran) [inline src/interface/web/app/components/navMenu/navMenu.tsx:176]: Let's revert to Settings here.

File: src/interface/web/app/factchecker/page.tsx [hunk 4]
```diff
@@ -533,7 +533,7 @@ export default function FactChecker() {
                         <Button disabled={clickedVerify} onClick={() => onClickVerify()}>Verify</Button>
                     </div>
                     <h3 className={`mt-4 mb-4`}>
-                        Try with a particular model. You must be <a href="/config" className="font-medium text-blue-600 dark:text-blue-500 hover:underline">subscribed</a> to configure the model.
+                        Try with a particular model. You must be <a href="/configure" className="font-medium text-blue-600 dark:text-blue-500 hover:underline">subscribed</a> to configure the model.
```
Comment (sabaimran) [inline src/interface/web/app/factchecker/page.tsx:536]: I think we should also change the page to be called settings. In `web_client.py`, we can temporarily add a redirect for people navigating to `/config` to go to `/settings`/.

File: src/khoj/routers/helpers.py [hunk 5]
```diff
@@ -1186,3 +1204,286 @@ def construct_automation_created_message(automation: Job, crontime: str, query_t
...
+    voice_model_options = list()
+    for voice_model in voice_models:
+        voice_model_options.append({"name": voice_model.name, "id": voice_model.model_id})
+
+    if len(voice_model_options) == 0:
+        eleven_labs_enabled = False
+
+    selected_voice_config = ConversationAdapters.get_voice_model_config(user)
+
+    return {
```
Comment (sabaimran) [inline src/khoj/routers/helpers.py:1273]: Oh man, lol.

File: src/khoj/routers/helpers.py [hunk 6]
```diff
@@ -1186,3 +1204,286 @@ def construct_automation_created_message(automation: Job, crontime: str, query_t
...
+    # Invalidate Query Cache
+    if user:
+        state.query_cache[user.uuid] = LRU()
+
+    return success
+
+
+NOTION_OAUTH_CLIENT_ID = os.getenv("NOTION_OAUTH_CLIENT_ID")
+NOTION_OAUTH_CLIENT_SECRET = os.getenv("NOTION_OAUTH_CLIENT_SECRET")
+NOTION_REDIRECT_URI = os.getenv("NOTION_REDIRECT_URI")
```
Comment (sabaimran) [inline src/khoj/routers/helpers.py:1483]: Move this to the top of the file? I generally like to keep environment variable imports at the top of files where possible.

File: src/khoj/routers/web_client.py [hunk 7]
```diff
@@ -199,115 +135,23 @@ def agent_page(request: Request, agent_slug: str):
...
-    paint_model_options = ConversationAdapters.get_text_to_image_model_options().all()
-    all_paint_model_options = list()
-    for paint_model in paint_model_options:
-        all_paint_model_options.append({"model_name": paint_model.model_name, "id": paint_model.id})
-
-    notion_oauth_url = get_notion_auth_url(user)
+    user_config = get_user_config(user, request, is_detailed=True)
 
-    eleven_labs_enabled = is_eleven_labs_enabled()
+    return templates.TemplateResponse("config.html", context=user_config)
```
Comment (sabaimran) [inline src/khoj/routers/web_client.py:147]: Nice!

File: src/khoj/routers/web_client.py [hunk 8]
```diff
@@ -199,115 +135,23 @@ def agent_page(request: Request, agent_slug: str):
...
-            "has_documents": has_documents,
-            "is_active": has_required_scope(request, ["premium"]),
-            "user_photo": user_picture,
-        },
-    )
+    return templates.TemplateResponse("agent.html", context=user_config)
 
 
-@web_client.get("/config", response_class=HTMLResponse)
+@web_client.get("/configure", response_class=HTMLResponse)
```
Comment (sabaimran) [inline src/khoj/routers/web_client.py:141]: Relevant code for renaming the page to `settings`.

```suggestion
@web_client.get("/settings", response_class=HTMLResponse)
```

File: src/khoj/routers/helpers.py (hunk 5 above)
Comment (debanjum) [reply src/khoj/routers/helpers.py:1278]: Yeah, there are a lot of them! I've arranged them with comment headers to make them a bit more readable

File: src/interface/web/app/factchecker/page.tsx (hunk 4 above)
Comment (debanjum) [reply src/interface/web/app/factchecker/page.tsx]: Yeah, I'd been shuttling between calling it configure vs settings. Configure makes it action oriented. Settings is more standard. I've renamed the user facing pages across web, desktop to use `Settings`. The API is still called `/api/configure` as the act of calling the API can be named with more action oriented semantics

File: src/interface/web/app/components/navMenu/navMenu.tsx (hunk 1 above)
Comment (debanjum) [reply src/interface/web/app/components/navMenu/navMenu.tsx]: Makes sense. Updated

**Pull request merged**